src/pages/studyView/StudyViewUtils.ts

```diff
@@ -63,3 +63,35 @@ export const COLORS = [
 export const NA_COLOR = '#CCCCCC'
 
 export const UNSELECTED_COLOR = '#808080'
+
+
+export function updateGeneQuery(geneQuery: string, selectedGene: string): string {
+
+    let isSelectedGenePresent = false;
+    let geneQueryLines = geneQuery.includes(':') ? geneQuery.split('\n') : _.isEmpty(geneQuery) ? [] : [geneQuery];
```

Comment:
I don't think this is safe, you should run OQL parser to get lines

Reply:
is there a way to get back the query string from OQL parser?. Currently OQL parser is just used to validate gene query. its result is not used anywhere

Reply:
I just made a PR to expose this function: https://github.com/cBioPortal/cbioportal-frontend/pull/1288